### PR TITLE
efi: minor tweaks to smbios parsing

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by anonymix007** [comment]
Probably the same changes need to be made in every place where `get_smbios_table` is called, that is, also in `smbios_in_hypervisor`

**Comment by poettering** [comment]
uh, so i was actually wrong about this, as it turns out get_smbios_table() already did the validation that the assert() then repeated. Hence the old code was fine.

Nonetheless, let's make this easier to read for morons like me, and move the full size validation into get_smbios_table(): both of the header and of the table size as a whole. And let's add an explanatory comment to the assert(), why it is an assert().

hence, sorry for the confusion, but with this new version of the PR I think we in the end are at a much better place. 

PTAL

File: src/boot/efi/vmm.c [hunk 1]
```diff
@@ -307,9 +307,10 @@ const char* smbios_find_oem_string(const char *name) {
 
         assert(left >= type11->header.length); /* get_smbios_table() already validated this */
         left -= type11->header.length;
+        const char *limit = s + left;
```
Comment (bluca) [inline src/boot/efi/vmm.c:310]: Nit, could be moved in the `for(` initialization field so that it's scoped? Doesn't matter that much

**Review by bluca — APPROVED** [review]

File: src/boot/efi/vmm.c (hunk 1 above)
Comment (yuwata) [reply src/boot/efi/vmm.c:310]: -> #34730.

**Pull request merged**